libs/cosmos-sdk/baseapp/baseapp_parallel.go

```diff
@@ -149,22 +148,14 @@ func (app *BaseApp) calGroup() {
 
 // ParallelTxs run txs
 func (app *BaseApp) ParallelTxs(txs [][]byte, onlyCalSender bool) []*abci.ResponseDeliverTx {
-	pm := app.parallelTxManage
-
 	txSize := len(txs)
-	pm.txSize = txSize
-	pm.haveCosmosTxInBlock = false
-	pm.workgroup.txs = txs
-	pm.isAsyncDeliverTx = true
-	pm.cms = app.deliverState.ms.CacheMultiStore()
-	pm.cms.DisableCacheReadList()
-	app.deliverState.ms.DisableCacheReadList()
-	pm.blockHeight = app.deliverState.ctx.BlockHeight()
 
 	if txSize == 0 {
 		return make([]*abci.ResponseDeliverTx, 0)
 	}
-	pm.init()
+
+	pm := app.parallelTxManage
+	pm.init(txs, app.deliverState.ctx.BlockHeight(), app.deliverState.ms)
 
 	app.getExtraDataByTxs(txs)
```

Comment:
getExtraDataByTxs inside can according to the cpu core start go routine

Reply:
done